components/browser/state/src/main/java/mozilla/components/browser/state/state/SessionState.kt

```diff
@@ -11,10 +11,12 @@ package mozilla.components.browser.state.state
  * @property content the [ContentState] of this session.
  * @property trackingProtection the [TrackingProtectionState] of this session.
  * @property engineState the [EngineState] of this session.
+ * @property extensions the list of extensions this [SessionState] has.
  */
 interface SessionState {
     val id: String
     val content: ContentState
     val trackingProtection: TrackingProtectionState
     val engineState: EngineState
+    val extensions: List<WebExtensionState> get() = emptyList()
```

Comment:
Don't need the `get()`?

Reply:
Without it the compiler get mad on us :(
<img width="897" alt="image" src="https://user-images.githubusercontent.com/773158/66532277-d80dd280-eadc-11e9-83e2-12e7c5fe08d2.png">

Reply:
Ah yes, but we don't need to set it to an `emptyList` in the interface. We initialize everything else in the impls `TabSessionState`, right?

Reply:
It was a bit convenient to do it there as `CustomTabSessionState` also inherit from  `SessionState` and  I didn't have to add it also on `CustomTabSessionState`.

I added it on `TabSessionState` to have a convenient parameter on the constructor, for the `createTab` function without it, I had to create `TabSessionState` object first and them set its `extensionState`.
  
![image](https://user-images.githubusercontent.com/773158/66536842-62a9fe00-eaec-11e9-9e9c-a013a43694a8.png)

**To keep consistency** between both `TabSessionState`and `CustomTabSessionState`. I will remove the initialization from the interface and initialized in both the child classes with  `emptyList`. Thanks for highlighting it!

Reply:
👍